### PR TITLE
Venmo Universal Link Return

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Add `BTVenmoClient(apiClient:universalLink:)` to use Universal Links when redirecting back from the Venmo flow
- Update Demo App
- Update Unit Tests
- Android PR: https://github.com/braintree/braintree_android/pull/1190

### Checklist

- [x] Added a changelog entry

### Authors

- @scannillo @jaxdesmarais

### CODE REVIEW

File: Sources/BraintreeVenmo/BTVenmoAppSwitchRedirectURL.swift
Comment (scannillo) [inline Sources/BraintreeVenmo/BTVenmoAppSwitchRedirectURL.swift:62]: Can we use URLComponents, or some other Apple built-in function, to do a path append? This way, we don't care if a merchant passes in a URL with an end ing slash `/` or not?

**Comment by scannillo** [comment]
I think URL schemes on the way back from Venmo was the only remaining need for a merchant to set `BTAppContextSwitcher.sharedInstance.returnURLScheme` 🥹. Let's deprecate it! 🔥

File: Sources/BraintreeVenmo/BTVenmoClient.swift
Comment (agedd) [inline Sources/BraintreeVenmo/BTVenmoClient.swift:53]: small nit (take it or leave it): 
```suggestion
    /// Creates a Venmo client
```

**Review by agedd — APPROVED** [review]
lgtm!! 🚀

**Review by richherrera — APPROVED** [review]

File: Sources/BraintreeCore/BTAppContextSwitcher.swift
Comment (scannillo) [inline Sources/BraintreeCore/BTAppContextSwitcher.swift:20]: Should this deprecation go in the CHANGELOG?

**Review by scannillo — APPROVED** [review]

File: Sources/BraintreeCore/BTAppContextSwitcher.swift
Comment (jaxdesmarais) [reply Sources/BraintreeCore/BTAppContextSwitcher.swift]: So I forgot about this swift limitation until now, but we cannot add this message since it causes cocoapods to fail ([previously discussed here](https://github.com/braintree/braintree_ios/pull/1130)). We cannot use the same solution in that PR because `BTAppContextSwitcher` lives in `Core` so the property would need to be public and the solution is to keep it internal. There may be some wonky workarounds but still no out of the box solutions. Any ideas?

Comment (scannillo) [reply Sources/BraintreeCore/BTAppContextSwitcher.swift]: We could do the same thing as that PR ^, but no-doc the "internal" (actually public) interior property that it references?

Comment (jaxdesmarais) [reply Sources/BraintreeCore/BTAppContextSwitcher.swift]: Ah, you're right, I had also added a deprecation message to the public property which resulted in the same error. 🤦

Comment (jaxdesmarais) [reply Sources/BraintreeCore/BTAppContextSwitcher.swift:20]: Updated: https://github.com/braintree/braintree_ios/pull/1440/commits/2705af71d15b7b0ee218ad97d9d27db7ff362c70

**Comment by jaxdesmarais** [comment]
Adding do not merge while we wait on RCIP approval for merge/release

**Comment by scannillo** [comment]
RCIP case RITM3021997 has been closed. This PR is ready for merge / release.

**Comment by jaxdesmarais** [comment]
Merging in and will be released once moratorium is lifted

**Pull request merged**
